app/src/main/java/com/bashkevich/androidfundamentals/model/workmanager/MovieLoaderWorker.kt

```diff
@@ -24,12 +25,22 @@ class MovieLoaderWorker(appContext: Context, params: WorkerParameters) : Corouti
     override suspend fun doWork(): Result = withContext(Dispatchers.IO) {
         try {
             val database = MoviesDatabase.getDatabase(applicationContext)
-            var nowPlayingMoviesIds = listOf<Int>()
-            var popularMoviesIds = listOf<Int>()
-            var topRatedMoviesIds = listOf<Int>()
-            var upcomingMoviesIds = listOf<Int>()
+            val sharedPreferences =
+                androidx.preference.PreferenceManager.getDefaultSharedPreferences(applicationContext)
+            var nowPlayingMoviesIds: Deferred<List<Int>>
+            var popularMoviesIds: Deferred<List<Int>>
+            var topRatedMoviesIds: Deferred<List<Int>>
+            var upcomingMoviesIds: Deferred<List<Int>>
             val allMoviesIds: List<Int>
 
+            val prevMovies = database.moviesDao.getAllMovies()
+
+            if (sharedPreferences.getInt(NEW_MOVIE_ID, 0) != 0) {
```

Comment:
Why remove it? Can't you just overwrite it after?

Reply:
I remove it in case of not showing movie in notification again if no new movies will be loaded